src/xdg.rs

```diff
@@ -262,6 +262,63 @@ impl IconLoader {
         Self { icons }
     }
 
+    fn premultiply_generic(data: &mut [u8]) {
+        // TODO: change to array_chunks_mut() once that is stabilised.
```

Comment:
It won’t be any faster I think, there are no checks here in the hot path, but it will avoid the need for the `if let` branch, and instead let us use `let`.